### ISSUE TITLE
lombok.Data annotation produces an error

### ISSUE THREAD

**Issue by dmitry-s** [google/error-prone#1081]
Please answer these questions before submitting your issue. Thanks!

### What version of Error Prone are you using?
2.2.0

### Does this issue reproduce with the latest release?
yes

### What did you do?
used lombok.Data annotation 

```
@Data
class TestDatastoreItem {
	private Duration durationField;

	private String stringField;

	private Boolean boolField;

	private Double doubleField;

	private Long longField;

	private LatLng latLngField;

	private Timestamp timestampField;

	private Blob blobField;

	private int intField;
}
```

### What did you expect to see?
I expected no error

### What did you see instead?
```
Compilation failure
[ERROR] /usr/local/google/home/dsolomakha/dev/spring-cloud-gcp/spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/TestDatastoreItem.java:[29,1] An unhandled exception was thrown by the Error Prone static analysis plugin.
[ERROR]      Please report this at https://github.com/google/error-prone/issues/new and include the following:
[ERROR]   
[ERROR]      error-prone version: 2.3.1
[ERROR]      BugPattern: ParameterName
[ERROR]      Stack Trace:
[ERROR]      java.lang.IndexOutOfBoundsException
[ERROR]   	at java.nio.HeapCharBuffer.subSequence(HeapCharBuffer.java:580)
[ERROR]   	at java.nio.HeapCharBuffer.subSequence(HeapCharBuffer.java:42)
[ERROR]   	at com.google.errorprone.bugpatterns.ParameterName.checkArguments(ParameterName.java:86)
[ERROR]   	at com.google.errorprone.bugpatterns.ParameterName.matchMethodInvocation(ParameterName.java:65)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitMethodInvocation(ErrorProneScanner.java:904)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitMethodInvocation(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1644)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
[ERROR]   	at com.sun.source.util.TreeScanner.visitUnary(TreeScanner.java:612)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitUnary(ErrorProneScanner.java:1123)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitUnary(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCUnary.accept(JCTree.java:1956)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
[ERROR]   	at com.sun.source.util.TreeScanner.visitIf(TreeScanner.java:418)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitIf(ErrorProneScanner.java:755)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitIf(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCIf.accept(JCTree.java:1427)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
[ERROR]   	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
[ERROR]   	at com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:248)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitBlock(ErrorProneScanner.java:527)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitBlock(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1026)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
[ERROR]   	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   	at com.sun.source.util.TreeScanner.visitMethod(TreeScanner.java:206)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:895)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:898)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
[ERROR]   	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
[ERROR]   	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
[ERROR]   	at com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:187)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:587)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
[ERROR]   	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
[ERROR]   	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
[ERROR]   	at com.sun.source.util.TreeScanner.visitCompilationUnit(TreeScanner.java:144)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:602)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:145)
[ERROR]   	at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:591)
[ERROR]   	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:64)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScannerTransformer.apply(ErrorProneScannerTransformer.java:43)
[ERROR]   	at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:152)
[ERROR]   	at com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:120)
[ERROR]   	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1404)
[ERROR]   	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1353)
[ERROR]   	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:952)
[ERROR]   	at com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:100)
[ERROR]   	at com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:142)
[ERROR]   	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:96)
[ERROR]   	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:90)
[ERROR]   	at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:137)
[ERROR]   	at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:108)
[ERROR]   	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:118)
[ERROR]   	at org.codehaus.plexus.compiler.javac.errorprone.JavacCompilerWithErrorProne$CompilerInvoker.compile(JavacCompilerWithErrorProne.java:223)
[ERROR]   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]   	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]   	at org.codehaus.plexus.compiler.javac.errorprone.JavacCompilerWithErrorProne.performCompile(JavacCompilerWithErrorProne.java:91)
[ERROR]   	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:943)
[ERROR]   	at org.apache.maven.plugin.compiler.TestCompilerMojo.execute(TestCompilerMojo.java:170)
[ERROR]   	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
[ERROR]   	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[ERROR]   	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[ERROR]   	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[ERROR]   	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]   	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]   	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
[ERROR]   	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]   	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
[ERROR]   	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
[ERROR]   	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
[ERROR]   	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
[ERROR]   	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
[ERROR]   	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
[ERROR]   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]   	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]   	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[ERROR]   	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[ERROR]   	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[ERROR]   	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

**Comment by cushon** [comment]
This is probably a duplicate of #780.

Error Prone expects to run on the AST prior to any desugaring, and Lombok generates synthetic AST nodes in a relatively early phase, so it's difficult to make the two compatible.

As a work-around you could disable that check with `-Xep:ParameterName:OFF` and see if that allows the compilation to succeed.
